lib/accessories/thermostat.js

```diff
@@ -115,31 +127,20 @@ FritzThermostatAccessory.prototype.setTargetHeatingCoolingState = function(state
         return;
     }
 
+    var service = this.services.Thermostat;
     this.platform.log(`Setting ${this.type} ${this.ain} heating state`);
-    var self = this, promise;
-
-    switch (state) {
-        case Characteristic.TargetHeatingCoolingState.OFF:
-        case Characteristic.TargetHeatingCoolingState.COOL:
-            promise = this.platform.fritz('getTempNight', this.ain);
-            break;
-        case Characteristic.TargetHeatingCoolingState.HEAT:
-            promise = this.platform.fritz('getTempComfort', this.ain);
-            break;
-        case Characteristic.TargetHeatingCoolingState.AUTO:
-            // let the Fritz!Box do its job
-            return;
-    }
 
-    promise.then(function(temp) {
-        if (temp === undefined) {
-            return;
-        }
 
-        self.platform.fritz('setTempTarget', self.ain, temp).then(function(temp) {
-            service.getCharacteristic(Characteristic.TargetTemperature).setValue(temp, undefined, FritzPlatform.Context);
-        });
-    });
+    let targetTemperatureString = 'off';
+    if (state === Characteristic.TargetHeatingCoolingState.OFF) {
+        targetTemperatureString = 'off';
+    } else if (state === Characteristic.TargetHeatingCoolingState.HEAT) {
+        targetTemperatureString = this.services.Thermostat.fritzTargetTemperature;
+    }
```

Comment:
Du änderst die komplette Logik. Meine war: OFF=Nachtmodus, ON=Komfortmodus.
Wenn Änderung notwendig (wofür?) dann bitte Erläuterung in PR und README.

Reply:
Bereits anfänglich erläutert und nun unten nochmal detailliert ausgeführt.